insights/parsers/xfs_info.py

```diff
@@ -47,18 +105,20 @@ def parse_content(self, content):
                     sect_info['specifier'] = match.group('specifier')
                     if match.group('specval'):
                         sect_info['specifier_value'] = match.group('specval')
-                for pair in self.keyval_re.findall(match.group('keyvaldata')):
+                for pair in keyval_re.findall(match.group('keyvaldata')):
                     (key, value) = pair
-                    if value[-1] != 's':
-                        # Value doesn't end with 'blks', so convert it to int.
+                    if value.isdigit():
+                        # Convert numeric values to integers as a convenience
                         value = int(value)
                     sect_info[key] = value
 
         self.xfs_info = xfs_info
 
         # Calculate a few things
-        self.data_size = xfs_info['data']['blocks'] * xfs_info['data']['bsize']
-        self.log_size = xfs_info['log']['blocks'] * xfs_info['log']['bsize']
+        if 'data' in xfs_info:
+            self.data_size = xfs_info['data']['blocks'] * xfs_info['data']['bsize']
```

Comment:
I think it would be safer to initialize them before the condition to avoid "AttributeError:" when the condition is not satisfied.

Reply:
Good point, will do.